### PR TITLE
Added a default constructor for torch.finfo.

### PR DESCRIPTION
**Pull request by benoitsteiner**

### CODE REVIEW

File: test/test_type_info.py [hunk 1]
```diff
@@ -39,6 +39,8 @@ def test_finfo(self):
             self.assertEqual(xinfo.max, xninfo.max)
             self.assertEqual(xinfo.eps, xninfo.eps)
             self.assertEqual(xinfo.tiny, xninfo.tiny)
+            torch.set_default_dtype(dtype)
```
[the remark this answers is not in the document]
Comment (benoitsteiner) [reply test/test_type_info.py]: Good idea, done.

File: docs/source/type_info.rst [hunk 2]
```diff
@@ -28,6 +28,9 @@ max         float   The largest representable number.
 tiny        float   The smallest positive representable number.
 =========   =====   ========================================
 
+.. note::
+  The constructor of :class:`torch.finfo` can be called without argument, in which case the class is created for the pytorch default type (as returned by ``torch.get_default_dtype()``).
```
Comment (gchanan) [inline docs/source/type_info.rst]: nit: 'default dtype' instead of 'default_type'

File: torch/csrc/TypeInfo.cpp [hunk 3]
```diff
@@ -50,16 +50,25 @@ PyObject* THPFInfo_pynew(PyTypeObject* type, PyObject* args, PyObject* kwargs) {
...
-    return PyErr_Format(
-        PyExc_TypeError,
-        "torch.finfo() requires a floating point input type. Use torch.iinfo to handle '%s'",
-        type->tp_name);
+  AT_CHECK(r.idx < 2, "Not a type");
+  at::ScalarType scalar_type;
+  if (r.idx == 1) {
+    scalar_type = torch::tensors::get_default_tensor_type().scalarType();
+    // The default tensor type can only be set to a floating point type/
+    AT_CHECK(at::isFloatingType(scalar_type));
```
Comment (gchanan) [inline torch/csrc/TypeInfo.cpp]: nit: AT_ASSERT instead of AT_CHECK.  AT_ASSERT is meant for internal checks.

**Review by gchanan — APPROVED** [review]
just a couple of nits.

File: torch/csrc/TypeInfo.cpp (hunk 3 above)
Comment (benoitsteiner) [reply torch/csrc/TypeInfo.cpp]: Done

File: docs/source/type_info.rst (hunk 2 above)
Comment (benoitsteiner) [reply docs/source/type_info.rst]: Done

**Pull request closed without merging**